### PR TITLE
Use -haddock for cabal and stack

### PR DESCRIPTION
**Pull request by jneira**
* After make sure with dont need `documentation: true` for sure
* This is related with #208 and #209, where i learned that hls/ghcide is using the docs included in ~~hie~~`.hi` files when you pass `-haddock` to ghc and no the docs generated by the haddock tool (with `documentation: true`)
  * See the mentioned issue/pr for more context

### CODE REVIEW

**Comment by georgefst** [comment]
To be pedantic, I *think* the info is in the `.hi` files, rather than `.hie` files.

File: cabal.project [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
          ghcide
 
 tests: true
-documentation: true
+
+program-options
+  ghc-options: -haddock
+
+package *
+  ghc-options: -haddock
```
Comment (pepeiborra) [inline cabal.project:11]: Why are two sets of `ghc-options` needed?

Comment (jneira) [reply cabal.project:11]: I think the second one should suffice but the fact is `cabal configure --ghc-options=-haddock` generates both in the `cabal.project.local`. Otoh hls/ghcide is parsing the local modules with haddock enabled in hls so docs are shown even if you dont set the `ghc-option` at all (maybe you can correct me if that is wrong, but i *think* we were having docs in hover for local modules before any `-haddock` around)
So the first one could be removed, i only was being consistent with `cabal configure` output.

File: stack-8.6.5.yaml [hunk 2]
```diff
@@ -4,6 +4,9 @@ packages:
 - .
 - ./ghcide/
 
+ ghc-options:
```
Comment (bubba) [inline stack-8.6.5.yaml:7]: I think you have an extra space in front of some of these

File: cabal.project (hunk 1 above)
Comment (jneira) [reply cabal.project:8]: Finally i've removed it, less is more

**Comment by jneira** [comment]
The stricter parser is throwed an error with hls itself:
```
Progress 33/35: haskell-language-serve  > /root/build/test/functional/Deferred.hs:162:9: error:
Progress 33/35: haskell-language-server  >     parse error on input ‘-- $ runSession hieCommand fullCaps "test/testdata" $ do’
Progress 33/35: haskell-language-server  >     |
Progress 33/35: haskell-language-server  > 162 |         -- $ runSession hieCommand fullCaps "test/testdata" $ do
Progress 33/35: haskell-language-server  >     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by jneira** [comment]
Parser errors solved so merging

**Pull request merged**